pingcap/tiflow
# feat(cdc): abort changefeed creation if there are running import task

### What problem does this PR solve?
Issue Number: close #8385

### What is changed and how it works?
feat(cdc): abort changefeed creation if there are running import task

### Check List 

#### Tests 


Assume we have full copy backup stored at /tmp/backup-1, log backup stored at /tmp/logbackup, and have a kafka cluster running at localhost:9092

1. setup tidb test cluster locally 
```
tiup playground nightly --tiflash 0
```
2. compiled CDC and start the server locally
```
./bin/cdc server
```
3. start the restore 
```
tiup br:nightly restore point --pd "localhost:2379" \
--storage='local:///tmp/logbackup' \
--full-backup-storage='local:///tmp/backup-1' \
--restored-ts '2023-03-16 16:59:57-0700'
```
4. at the meantime, try to create a changefeed
```
./bin/cdc cli changefeed create \
--pd=http://127.0.0.1:2379 \
--changefeed-id="simple-replication-task" \
--sort-engine="unified" \
--sink-uri="kafka://127.0.0.1:9092/quickstart-events?protocol=canal-json" 
```
and we will see the error message
`Error: There are lightning/restore tasks runningplease stop or wait for them to finish. If the lightning/restore task is terminated by system, please wait until the task lease ttl(3 mins) decreases to 0.`


#### Questions 

##### Will it cause performance regression or break compatibility?
No

##### Do you need to update user documentation, design documentation or monitoring documentation?
No

### Release note 

```release-note
`None`
```
